src/xtgeo/xyz/polygons.py

```diff
@@ -261,6 +237,32 @@ def dataframe(self):
     @dataframe.setter
     def dataframe(self, df):
         self._df = df.copy()
+        self._check_column_name()
+
+    def _check_column_name(self):
+        if self._dtname not in self._df.columns:
+            self._dtname = None
+        if self._dhname not in self._df.columns:
+            self._dhname = None
+        if self._tname not in self._df.columns:
+            self._tname = None
+        if self._hname not in self._df.columns:
+            self._hname = None
+
+    def _validify_name(self, value):
+        if not isinstance(value, str):
+            raise ValueError(
+                "Wrong type of input; must be string, was {}".format(
+                    type(value)
+                )
+            )
+
+        if value not in self._df.columns:
+            raise ValueError(
+                "{} does not exist as a column name, must be one of: {}".format(
+                    value, self._df.columns
```

Comment:
adding the list of possible column names might be too much noise, I'm not familiar with the usage (or the input dataframe coming from the user)..